[김성주] sprint3

626-ju opened the pull request:
# 요구사항
https://panda-sprint3.netlify.app/

## 기본

   #### 공통
브라우저에 현재 보이는 화면의 영역(viewport) 너비를 기준으로 분기되는 반응형 디자인을 적용합니다.
- [x] PC: 1200px 이상
- [x] Tablet: 768px 이상 ~ 1199px 이하
- [x] Mobile: 375px 이상 ~ 767px 이하
375px 미만 사이즈의 디자인은 고려하지 않습니다

#### 랜딩 페이지
- [x] Tablet 사이즈로 작아질 때 “판다마켓” 로고의 왼쪽에 여백 24px, “로그인” 버튼 오른쪽 여백 24px을 유지할 수 있도록 “판다마켓” 로고와 “로그인" 버튼의 간격이 가까워집니다.
- [x]  Mobile 사이즈로 작아질 때 “판다마켓” 로고의 왼쪽에 여백 16px, “로그인” 버튼 오른쪽 여백 16px을 유지할 수 있도록 “판다마켓” 로고와 “로그인" 버튼의 간격이 가까워집니다.
- [x] 화면 영역이 줄어들면 “Privacy Policy”, “FAQ”, “codeit-2024”이 있는 영역과 SNS 아이콘들이 있는 영역의 간격이 줄어듭니다.

#### 로그인, 회원가입 페이지 공통
Tablet 사이즈에서 내부 디자인은 PC사이즈와 동일합니다.
- [x] Mobile 사이즈에서 좌우 여백 16px 제외하고 내부 요소들이 너비를 모두 차지합니다.
- [x] Mobile 사이즈에서 내부 요소들의 너비는 기기의 너비가 커지는 만큼 커지지만 400px을 넘지 않습니다.
## 심화

- [x] 페이스북, 카카오톡, 디스코드, 트위터 등 SNS에서 Linkbrary 랜딩 페이지(“/”) 공유 시 좌측 예시와 같은 미리보기를 볼 수 있도록 랜딩 페이지 메타 태그를 설정해 주세요.
- [x] 미리보기에서 제목은 “판다 마켓”, 설명은 “일상의 모든 물건을 거래해보세요”로 설정합니다.
주소와 이미지는 자유롭게 설정하세요.

## 주요 변경사항
- @media 사용으로 분기 나눠서 다른 스타일 적용
- 디자인이 바뀔 때 중복된 속성을 재정의하는 경우가 잦아 중복된 속성들을 변수처리하고 파일을 분리했습니다.

## 스크린샷
(랜딩 모바일)
![모바일 1](https://github.com/user-attachments/assets/51eff316-e151-4ebe-8d88-c3b8c2699c7e)


<hr>

(랜딩 태블릿)
![태블릿 1](https://github.com/user-attachments/assets/16f93b06-f03b-4e13-8b64-1ed544e5e3ee)

<hr>

(랜딩 데스크탑)
![pc](https://github.com/user-attachments/assets/907ca23a-35b1-4fbe-8350-743e7682e28d)
<hr>

(로그인 모바일)
![로그인_모바일](https://github.com/user-attachments/assets/ab678333-2c50-40c8-840d-23ade100f7e1)
<hr>

(회원가입 모바일)
![회원가입_모바일](https://github.com/user-attachments/assets/c6f64e8b-f5ff-40fa-b38d-0b8cd23aff5c)
<hr>

![캡처](https://github.com/user-attachments/assets/c664aaab-f485-4d29-b623-b523aaf59a9e)

## 멘토에게

-
-
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

626-ju committed:
@media 추가

626-ju committed:
global.css수정

1)body에 word-break:keep-all
2) --footer-text 컬러변수 추가

626-ju committed:
og주소 변경

626-ju committed:
og 이미지 수정

626-ju committed:
op이미지경로 웹상 경로로

626-ju committed:
css변수화

626-ju committed:
반응형-변수파일 분리

article 내부 h2 letter-spacing2% => 0.02em

626-ju committed:
포멧팅수정

addiescode-sj commented on global.css:
```diff
@@ -27,4 +28,5 @@ body {
     --secondary-gray200: #E5E7EB;
     --secondary-gray100: #F3f4f6;
     --secondary-gray50: #f9fafb;
+    --footer-text:#CFCFCF;
```
포맷팅 조금 더 신경써봐요! 

```suggestion
    --footer-text: #CFCFCF;
```

addiescode-sj commented on index.html:
```diff
@@ -40,7 +57,7 @@ <h2>일상의 모든 물건을<br> 거래해 보세요</h2>
                         <span>Hot item</span>
                         <h2>
-                            인기 상품을<br>
+                            인기 상품을
```
그래도 디자인 의도를 충분히 반영하기위해 디자인상 줄바꿈은 그대로 유지하시고,
화면 크기에 따른 유연한 줄바꿈을 위해 word-break와 같은 속성을 써주시는게 좋아요.

addiescode-sj commented on login/login.css:
```diff
@@ -113,4 +117,36 @@ button:hover {
   color: var(--primary-color);
+}
+
+@media screen and (min-width:768px) {
```
굳굳! 반응형 레이아웃 잘 구성하셨네요 :)

addiescode-sj commented on responsive-variables.css:
컨테이너 요소의 레이아웃을 결정하는 주요 속성 (e.g 여백, 크기, 간격) 들만 CSS 변수로 관리해보는건 어떨까요?
그리고 해당 파일을 따로 분리하지않고, 컴포넌트 레벨 (페이지별 CSS) 파일에 사용되었으면 좋겠어요 . 
(디자인을 수정하거나 추가할때 로컬에서만 쓰이는 CSS로 디버깅 범위가 한정되지않고, 이 CSS파일에 있는 다른 스타일 규칙들도 참고해야해서 관리에 어려움이 있을수있습니다 :))

addiescode-sj commented on style.css:
```diff
@@ -1,279 +1,396 @@
-/* Main */
-section.main__hero {
+/* main 공통 */
+section>article img {
```
여기서도 띄어쓰기 한칸씩 해줍시다! 
포맷팅할때 기준이 햇갈리신다면, prettier 기본 규칙 참고해서 좋은 포맷팅을 유지할수있도록 해볼까요?

[참고](https://velog.io/@2_hyeonju/Prettier-ESLint-%EC%99%84%EC%A0%84-%EC%A0%95%EB%B3%B5)

```suggestion
section > article img {
```

addiescode-sj reviewed (approved):
수고하셨습니다!

## 주요 리뷰 포인트
- 미디어쿼리안에서 CSS 변수 사용 관련 피드백
- 포맷팅
- CSS 파일 분리 관련 피드백

626-ju committed:
refactor(mentor): 포멧팅 수정 및 변수 관리

1) prettier 기본 규칙 준수해서 포멧팅 수정하기
2) 여백, 크기, 간격 관련된 것만 변수로 관리(font-size, line-height
   같은 건 빼기)
3) 반응형에서 사용하는 변수들 같은 파일에서 관리하기

626-ju committed:
fix: 줄바꿈 조절
